### DISCUSSION TITLE
How to verify if the Webhook call came from Swell?

### DISCUSSION THREAD

**Discussion by jramoyo-serenade** [swellstores/community#347] [Help] [answered]
How can we verify if a Webhook call came from Swell?

We were hoping we can use the "Alias" to set a shared secret -- but it does not seem to get sent back with the Webhook call.

![Screenshot 2023-08-17 at 11 58 02 am](https://github.com/swellstores/community/assets/133632539/3a36237c-0014-4959-8cd9-266724db0378)

Headers:

<img width="615" alt="Screenshot 2023-08-17 at 11 59 24 am" src="https://github.com/swellstores/community/assets/133632539/4302ccd8-9c93-4ad2-94bb-2e2514136f35">

**Comment by logeshswell** [comment, marked as answer]
Currently, we don't have any security mechanism to verify the webhook data whether it is triggered from Swell.
Instead, we recommend fetching the associated record using `data.id` before performing a relevant action for the time being.
https://developers.swell.is/backend-api/webhooks/receiving-webhooks

**Reply by jramoyo-serenade** [reply]
Thanks for the response, do you have plans to implement this in the near future? 

This approach is counter-intuitive and wouldn't this impact API limits?

**Comment by ericingram** [comment]
We typically recommend putting a shared secret in the webhook URL itself

**Comment by logeshswell** [comment]
@jramoyo-serenade You can also verify it using the list of IP addresses from which we will trigger webhooks:

```
52.52.111.237
54.219.85.17
54.241.235.166
```

Would this work for you?

**Reply by ruanmatthee-work** [reply]
@logeshswell How often would these change? And where/how can we check for changes?

**Reply by logeshswell** [reply]
IPs are very less likely to be changed. ~But, we don't have any medium to get to know if it's changed.~

**Reply by logeshswell** [reply]
We can update the IP list in the below page if it's changed.
https://developers.swell.is/backend-api/webhooks/receiving-webhooks
